tests/unit/mathematics/linalg/RationalApproximation_unittest.cc

```diff
@@ -136,50 +126,22 @@ TEST(RationalApproximation, trace_accuracy)
 
 	// create the operator function that extracts the trace
 	// of the approximation of log of the linear operator
-	CLogRationalApproximationIndividual *op_func
-	=new CLogRationalApproximationIndividual(
-			op, e, eig_solver, linear_solver, accuracy);
+	CLogRationalApproximationIndividual* op_func =
+	    new CLogRationalApproximationIndividual(
+	        op, eig_solver, linear_solver, accuracy);
 	SG_REF(op_func);
 
 	op_func->precompute();
-
-	// create the aggregators to contain the result aggregators
-	CDynamicObjectArray aggregators;
-
+	float64_t result = 0.0;
 	// extract the trace of approximation of log using basis vectors
 	for (index_t i=0; i<size; ++i)
 	{
 		SGVector<float64_t> s(size);
 		s.set_const(0.0);
 		s[i]=1.0;
-		CJobResultAggregator* agg=op_func->submit_jobs(s);
-		aggregators.append_element(agg);
-		SG_UNREF(agg);
-	}
-	// wait for all computation jobs to be computed
-	e->wait_for_all();
-
-	// use the aggregators to find the final result
-	int32_t num_aggregates=aggregators.get_num_elements();
-	float64_t result=0.0;
-	for (int32_t i=0; i<num_aggregates; ++i)
-	{
-		CJobResultAggregator* agg=dynamic_cast<CJobResultAggregator*>
-			(aggregators.get_element(i));
-
-		// call finalize on all the aggregators
-		agg->finalize();
-		CScalarResult<float64_t>* r=dynamic_cast<CScalarResult<float64_t>*>
-			(agg->get_final_result());
-
-		// its important that we don't just unref the result here
-		result+=r->get_result();
-		SG_UNREF(agg);
+		result += op_func->compute(s);
```

Comment:
it seems you are not changing anything here ... don't understand why the test fails.

Reply:
yeah i am confused too... also the RationalApproximationCGM tests pass its the one with 
RationalApproximationIndividual that fail.

Reply:
you will have to carefully backtrace the changes.
If staring at the code doesnt help, you can run the two implementations next to each other in a debugger and ensure that the same number is added in each step. Ping me if you need help with this, I am in IRC for another hour

Reply:
okay i am on it.